src/pytest_plugins/consume/simulators/base.py

```diff
@@ -0,0 +1,86 @@
+"""Common pytest fixtures for the Hive simulators."""
+
+from pathlib import Path
+from typing import Dict, Literal
+
+import pytest
+from hive.client import Client
+
+from ethereum_test_fixtures import (
+    BaseFixture,
+)
+from ethereum_test_fixtures.consume import TestCaseIndexFile, TestCaseStream
+from ethereum_test_fixtures.file import Fixtures
+from ethereum_test_rpc import EthRPC
+from pytest_plugins.consume.consume import FixturesSource
+
+
+@pytest.fixture(scope="function")
+def eth_rpc(client: Client) -> EthRPC:
+    """Initialize ethereum RPC client for the execution client under test."""
+    return EthRPC(f"http://{client.ip}:8545")
+
+
+@pytest.fixture(scope="function")
+def check_live_port(test_suite_name: str) -> Literal[8545, 8551]:
+    """Port used by hive to check for liveness of the client."""
+    if test_suite_name == "eest/consume-rlp":
+        return 8545
+    elif test_suite_name == "eest/consume-engine":
+        return 8551
+    raise ValueError(
+        f"Unexpected test suite name '{test_suite_name}' while setting HIVE_CHECK_LIVE_PORT."
+    )
+
+
+class FixturesDict(Dict[Path, Fixtures]):
```

Comment:
I see that this is a copy/paste from the file above, but I'm curious about this caching just for my own knowledge base. Does the built in pytest fixture scoping fall short?

Reply:
JSON fixture files are loaded upon demand by the simulator. The simulator knows which JSON fixture file contains the current test case (test fixture) from the [index file] (https://eest.ethereum.org/main/running_tests/consume/cache/?h=index+file#the-fixture-index-file). 

A JSON fixture file can contain multiple fixtures, so this avoids unnecessarily loading the JSON multiple times. Due to the lack of a 1-to-1 mapping it doesn't line up well with pytest fixture scoping. 

Make sense? It's really unfortunate that we have two meanings for fixture 😆